src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1921,18 +1941,18 @@ bool isRefEscape
                     _ => throw ExceptionUtilities.UnexpectedValue(symbol)
                 };
 
-                if (receiver is not BoundValuePlaceholderBase && method is not null && receiver.Type!.IsValueType)
+                if (receiver is not BoundValuePlaceholderBase && method is not null && receiver.Type?.IsValueType == true)
```

Comment:
Was there a bug here (with `null` `receiver.Type`)?

Reply:
This did fail at one point while making these changes but doesn't appear to fail anymore. That said, it feels unnecessary to assume `receiver.Type` is not `null`.

Reply:
I don't feel strongly either way, but it doesn't feel right to me that you could be invoking something on a receiver, yet the receiver lacks a type.